File: Java/프로그래머스/1/340199. ［PCCE 기출문제］ 9번 ／ 지폐 접기/［PCCE 기출문제］ 9번 ／ 지폐 접기.java
class Solution {
    public int solution(int[] wallet, int[] bill) {
        int answer = 0;
       // wallet과 bill을 한번만 정렬
        //sort(wallet);
        //sort(bill);
        
        // 지갑에 지폐가 들어갈 때까지 반복
        while (!check(wallet, bill)) {
            rebill(bill);  // 지폐 크기를 줄임
            answer++;
        }
        
        return answer;
    }

    // 배열을 오름차순으로 정렬 (복사 없이 제자리 정렬)
    public void sort(int[] num) {
        if (num[0] > num[1]) {
            int temp = num[0];
            num[0] = num[1];
            num[1] = temp;
        }
    }

    // 지갑에 지폐가 들어갈 수 있는지 확인
    public boolean check(int[] wallet, int[] bill) {
        return wallet[0] >= bill[0] && wallet[1] >= bill[1] || wallet[0] >= bill[1] && wallet[1] >= bill[0];
    }

    // 더 큰 지폐의 크기를 반으로 줄임
    public void rebill(int[] bill) {
        if (bill[0] > bill[1]) {
            bill[0] /= 2;
        } else {
            bill[1] /= 2;
        }
    }
}

File: Python3/프로그래머스/1/340199. ［PCCE 기출문제］ 9번 ／ 지폐 접기/［PCCE 기출문제］ 9번 ／ 지폐 접기.py
def solution(wallet, bill):
    answer = 0
    # sort(reverse=True) 내림차순 (큰값부터 정렬)
    wallet.sort(reverse=True)  
    bill.sort(reverse=True)
    
    while 1:
        # 지폐와 지갑의 크기를 비교후 지폐가 크면 반으로 접음
        # 지폐가 지갑보다 값이 작아지면 반복문 멈추게
        if bill[0] > wallet[0] or bill[1] > wallet[1]:

            bill[0] = int(bill[0] / 2)
            # 다시한번 내림차순으로 정렬하여 큰값이 앞으로 오게
            bill.sort(reverse=True)
            answer += 1
        else:
            break
    return answer